### PR TITLE
[#11] [iOS] [UI] As a user, I can reset my password

### LINKED ISSUE #11: [iOS] [UI] As a user, I can reset my password
## Why

When user forgot their password, they should be able to reset it.

## Acceptance Criteria?

- Show logo.
- Show instruction.
- Show email textfield.
- Show `Reset` button.

- Show loading indicator.

## Resources

Screen Shot 2565-09-12 at 16 58 14

[Postman - Nimble Survey API](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A457)

### PR DESCRIPTION
**Pull request by blyscuit**
close #11

## What happened

Add Reset Password view.

## Insight

- Add loading indicator overlay.
- Add back button customizer to disable back button when indicator overlay is showing.
- Use back button customizer to allow swipe back.

## Proof of Work

https://user-images.githubusercontent.com/6356137/192445191-d3273d52-b117-414c-bd71-f2f0606eeaf3.mp4

https://user-images.githubusercontent.com/6356137/192445221-c6564965-f668-4121-a2be-5735af245cac.mp4

### CODE REVIEW

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift [hunk 1]
```diff
@@ -16,44 +21,53 @@ struct LoginView: View {
...
+                    .ignoresSafeArea()
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+
+                Assets.backgroundBlur
+                    .image
+                    .resizable()
+                    .aspectRatio(contentMode: .fill)
+                    .ignoresSafeArea()
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+                    .opacity(animating ? 1.0 : 0.001)
```
Comment (minhnimble) [inline iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift]: Can't we set it to `0.0` when not animating? 🤔

File: iosApp/Survey/Sources/Presentation/Modules/ResetPassword/ResetPasswordView.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+                        Text(Localize.resetPasswordTextInstruction())
+                            .multilineTextAlignment(.center)
+
+                        Spacer().frame(maxHeight: 70.0)
+
+                        emailField
+                        resetButton
+                    }
+                    .padding(.horizontal, 24.0)
```
Comment (minhnimble) [inline iosApp/Survey/Sources/Presentation/Modules/ResetPassword/ResetPasswordView.swift]: The more of these padding/spacing values will be used in the application the more screens we have. So I suggest you to create constants values for these for later reusing and modification such as `smallPadding`, `mediumPadding`, `largePadding`, etc. What do you think?

File: iosApp/Survey/Sources/Supports/Extensions/SwiftUI/View+BackButton.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+extension View {
+
+    func defaultBackButton(_ presentationMode: Binding<PresentationMode>) -> some View {
+        var backButton: some View {
+            Button {
+                presentationMode.wrappedValue.dismiss()
+            } label: {
+                HStack(spacing: 0) {
+                    Image(systemName: "chevron.left")
```
Comment (minhnimble) [inline iosApp/Survey/Sources/Supports/Extensions/SwiftUI/View+BackButton.swift]: I think we should create constants for system image strings also for better management and potential later usage 🙏

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift [hunk 4]
```diff
@@ -62,19 +76,19 @@ struct LoginView: View {
     }
 
     var loginField: some View {
-        TextField(Localize.loginFieldsEmail(), text: $email)
+        TextField(Localize.loginFieldEmail(), text: $email)
             .keyboardType(.emailAddress)
             .primaryTextField()
             .accessibility(.login(.emailField))
     }
```
Comment (minhnimble) [inline iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift:83]: I think we should name this View to `emailField` for consistency 🙏 Btw, what do you think about extracting the `loginField` to a new file so it they can be shared between the `Login` and `ForgotPassword` screen?  🤔

Comment (blyscuit) [reply iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift:83]: I think sharing the style through `primaryTextField` is enough for this view?

File: iosApp/Survey/Sources/Supports/Extensions/SwiftUI/View+BackButton.swift (hunk 3 above)
Comment (blyscuit) [reply iosApp/Survey/Sources/Supports/Extensions/SwiftUI/View+BackButton.swift]: Added in https://github.com/blyscuit/kmm-swiftUI-IC/pull/56/commits/8edbbc2a81db456ef46a1258d58b109b6a24d252

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift (hunk 1 above)
Comment (blyscuit) [reply iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift]: Added in https://github.com/blyscuit/kmm-swiftUI-IC/pull/56/commits/8edbbc2a81db456ef46a1258d58b109b6a24d252

File: iosApp/Survey/Sources/Presentation/Modules/ResetPassword/ResetPasswordView.swift (hunk 2 above)
Comment (blyscuit) [reply iosApp/Survey/Sources/Presentation/Modules/ResetPassword/ResetPasswordView.swift]: Added in https://github.com/blyscuit/kmm-swiftUI-IC/pull/56/commits/8edbbc2a81db456ef46a1258d58b109b6a24d252

File: iosApp/Survey/Sources/Presentation/Views/EmbedLoadingView.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+//  EmbedLoadingView.swift
+//  Survey
+//
+//  Created by Bliss on 27/9/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import SwiftUI
+
+struct EmbedLoadingView<Content: View>: View {
```
Comment (suho) [inline iosApp/Survey/Sources/Presentation/Views/EmbedLoadingView.swift]: How about using `ViewModifier` in this case?

I read this library: https://github.com/globulus/swiftui-generic-dialog/blob/main/Sources/SwiftUIGenericDialog/SwiftUIGenericDialog.swift

So for loading, we can have the same approach

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift (hunk 4 above)
Comment (minhnimble) [reply iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift:83]: @blyscuit okay sure, just an optional suggestion at this point. We can definitely revisit if the sharing needs is more imminent 🙏

**Review by minhnimble — APPROVED** [review]
Apart from @suho comment, rest LGTM 🚀

File: iosApp/Survey/Sources/Presentation/Views/EmbedLoadingView.swift (hunk 5 above)
Comment (blyscuit) [reply iosApp/Survey/Sources/Presentation/Views/EmbedLoadingView.swift]: I added the approach you suggested, looks nice. https://github.com/blyscuit/kmm-swiftUI-IC/pull/56/commits/d42d493ec1a297624b8965dde89ca09c4db30730

**Review by suho — APPROVED** [review]

**Pull request merged**